js/ethereum-utils/src/signature.ts

```diff
@@ -33,49 +36,76 @@ import {
   PAGINATED_DELETE_SIGNATURE_PAYLOAD_DEFINITION_V2,
   PAGINATED_UPSERT_SIGNATURE_PAYLOAD_DEFINITION_V2,
   PASSKEY_PUBLIC_KEY_DEFINITION,
+  SIWF_SIGNED_REQUEST_PAYLOAD_DEFINITION,
   SupportedPayloadDefinitions,
-} from './signature.definitions';
+} from './signature.definitions.js';
 import { KeyringPair } from '@polkadot/keyring/types';
 import { Signer, SignerResult } from '@polkadot/types/types';
 
 /**
- * Signing EIP-712 compatible signature for payload
+ * Signing EIP-712 or ERC-191 compatible signature based on payload
  * @param secretKey
  * @param payload
  * @param chain
  */
-export async function signEip712(
+export async function sign(
   secretKey: HexString,
   payload: SupportedPayload,
   chain: ChainType = 'Mainnet-Frequency'
 ): Promise<EcdsaSignature> {
-  const types = getTypesFor(payload.type);
+  const signatureType = getSignatureType(payload.type);
   const normalizedPayload = normalizePayload(payload);
   const wallet = new ethers.Wallet(secretKey);
-  // TODO: use correct chainID for different networks
-  // TODO: use correct contract address for different payloads
-  const signature = await wallet.signTypedData(EIP712_DOMAIN_DEFAULT, types, normalizedPayload);
+  let signature;
+  switch (signatureType) {
+    case 'EIP-712':
+      // TODO: use correct chainID for different networks
+      // TODO: use correct contract address for different payloads
```

Comment:
We really should figure out these todos. I know the chainID we can get setup, but less sure about what contract address we should use as we will not have those until we have the full pallet->contract precompiles setup.

Any ideas on what it would take to alter these values? I guess we'd have to support both for a while.

Reply:
We decided to keep it as is for now.